opentelemetry-exporter-geneva/geneva-uploader/src/config_service/client.rs

```diff
@@ -0,0 +1,536 @@
+    MonikerNotFound(String),
+    #[error("Internal error: {0}")]
+    InternalError(String),
+    //#[error("Mismatched TagId. Sent: {sent}, Received: {received}")]
+    //MismatchedTagId { sent: String, received: String },
+}
+
+#[allow(dead_code)]
+pub(crate) type Result<T> = std::result::Result<T, GenevaConfigClientError>;
+
+/// Configuration for the Geneva Config Client.
+///
+/// # Fields
+/// * `endpoint` - The Geneva Config Service endpoint URL
+/// * `environment` - Environment name (e.g., "prod", "dev")
+/// * `account` - Account name in Geneva
+/// * `namespace` - Namespace for the configuration
+/// * `region` - Azure region (e.g., "westus2")
+/// * `config_major_version` - Major version of the configuration schema
+/// * `auth_method` - Authentication method to use (Certificate or ManagedIdentity)
+///
+/// # Example
+/// ```ignore
+/// let config = GenevaConfigClientConfig {
+///     endpoint: "https://example.geneva.com".to_string(),
+///     environment: "prod".to_string(),
+///     account: "myaccount".to_string(),
+///     namespace: "myservice".to_string(),
+///     region: "westus2".to_string(),
+///     config_major_version: 1,
+///     auth_method: AuthMethod::Certificate {
+///         path: "/path/to/cert.p12".to_string(),
+///         password: "password".to_string(),
+///     },
+/// };
+/// ```
+#[allow(dead_code)]
+#[derive(Clone)]
+pub(crate) struct GenevaConfigClientConfig {
+    pub endpoint: String,
+    pub environment: String,
+    pub account: String,
+    pub namespace: String,
+    pub region: String,
+    pub config_major_version: u32,
+    pub auth_method: AuthMethod, // agent_identity and agent_version are hardcoded for now
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Deserialize)]
+pub(crate) struct IngestionGatewayInfo {
+    #[serde(rename = "Endpoint")]
+    pub(crate) endpoint: String,
+    #[serde(rename = "AuthToken")]
+    pub(crate) auth_token: String,
+    #[serde(rename = "AuthTokenExpiryTime")]
+    pub(crate) auth_token_expiry_time: String,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub(crate) struct MonikerInfo {
+    pub name: String,
+    pub account_group: String,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Deserialize)]
+struct StorageAccountKey {
+    #[serde(rename = "AccountMonikerName")]
+    account_moniker_name: String,
+    #[serde(rename = "AccountGroupName")]
+    account_group_name: String,
+    #[serde(rename = "IsPrimaryMoniker")]
+    is_primary_moniker: bool,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Deserialize)]
+struct GenevaResponse {
+    #[serde(rename = "IngestionGatewayInfo")]
+    ingestion_gateway_info: IngestionGatewayInfo,
+    // Make storage_account_keys optional since it might not be present in all responses
+    #[serde(rename = "StorageAccountKeys", default)]
+    storage_account_keys: Vec<StorageAccountKey>,
+    // Keep tag_id as it might be used for validation
+    #[serde(rename = "TagId", default)]
+    tag_id: Option<String>,
```

Comment:
Is tag_id also not always expected to be a part of the response?

Reply:
yes, made it non-optional parameter.